packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/InAppPurchaseStoreKit2.swift

```diff
@@ -34,4 +34,150 @@ extension InAppPurchasePlugin: InAppPurchase2API {
       }
     }
   }
+
+  /// Gets the appropriate product, then calls purchase on it.
+  /// https://developer.apple.com/documentation/storekit/product/3791971-purchase
+  func purchase(
+    id: String, options: SK2ProductPurchaseOptionsMessage?,
+    completion: @escaping (Result<SK2ProductPurchaseResultMessage, Error>) -> Void
+  ) {
+    Task {
+      @MainActor in
+      do {
+        let product = try await Product.products(for: [id]).first
+        guard let product = product else {
+          let error = PigeonError(
+            code: "storekit2_failed_to_fetch_product",
+            message: "Storekit has failed to fetch this product.",
+            details: "Product ID : \(id)")
+          return completion(.failure(error))
+        }
+
+        let result = try await product.purchase(options: [])
+
+        switch result {
+        case .success(let verification):
+          switch verification {
+          case .verified(let transaction):
+            self.sendTransactionUpdate(transaction: transaction)
+            completion(.success(result.convertToPigeon()))
+          case .unverified(_, let error):
+            completion(.failure(error))
+          }
+        case .pending:
+          completion(
+            .failure(
```

Comment:
reduce indentation by combining these

Reply:
the formatter uncombines these

Reply:
strange. it's clearly less than 100 characters. how's your code like before the formatter?